pkg/changelog/changelog.go

```diff
@@ -0,0 +1,254 @@
+package changelog
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/giantswarm/microerror"
+)
+
+// Regex patterns used by all Giant Swarm components
+const (
+	commonStartPattern = "(?m)^## \\[v?(?P<Version>\\d+\\.\\d+\\.\\d+)\\].*(?P<Date>\\d{4}-\\d{2}\\-\\d{2})$"
+	commonEndPattern   = "(?m)^\\[.*\\]:.*$"
+)
+
+// Parameters defining how to parse and extract release info about all known components
+var knownComponentParseParams = map[string]parseParams{
```

Comment:
Fine for now but this should live somewhere easily findable.

Reply:
Agreed. We can probably have a few generic parsers that handle different styles of changelog. Keep a Changelog-style, Kubernetes-style, Container Linux-style, etc.